anndata/tests/test_io_elementwise.py

```diff
@@ -302,6 +305,57 @@ def test_read_zarr_from_group(tmp_path, consolidated):
     assert_equal(adata, expected)
 
 
+# Test dask sparse array IO above int32 size limit
+
+
+class CSRCallable:
+    """Dummy class to bypass dask checks"""
+
+    def __new__(cls, shape, dtype) -> sparse.csr_matrix:
+        if len(shape) == 0:
+            shape = (0, 0)
+        if len(shape) == 1:
+            shape = (shape[0], 0)
+        elif len(shape) == 2:
+            pass
+        else:
+            raise ValueError(shape)
+        return sparse.csr_matrix(shape, dtype=dtype)
+
+
+def make_dask_chunk(
+    x: ad.experimental.SparseDataset, start: int, end: int
+) -> DaskArray:
+    import dask.array as da
+    from dask import delayed
+
+    def take_slice(x, idx):
+        return x[idx]
+
+    return da.from_delayed(
+        delayed(take_slice)(x, slice(start, end)),
+        dtype=x.dtype,
+        shape=(end - start, x.shape[1]),
+        meta=CSRCallable,
+    )
+
+
+def sparse_dataset_as_dask(x, stride: int):
+    import dask.array as da
+
+    n_chunks, rem = divmod(x.shape[0], stride)
+
+    chunks = []
+    cur_pos = 0
+    for i in range(n_chunks):
+        chunks.append(make_dask_chunk(x, cur_pos, cur_pos + stride))
+        cur_pos += stride
+    if rem:
+        chunks.append(make_dask_chunk(x, cur_pos, x.shape[0]))
+
+    return da.concatenate(chunks, axis=0)
+
+
```

Comment:
> Can we move to helpers so we can import in scanpy?

We can't import from scanpy since the scanpy user may have an older version of anndata.

> Also this does not appear to be used anywhere?

This should probably exist in a utils somewhere, and get tested, but not totally sure where at the moment...

Could remove for now?

Reply:
> We can't import from scanpy since the scanpy user may have an older version of anndata. 

For the future?

> Maybe it just exists in some docs?

This seems like a must, great idea, but would not be opposed to testing as well.

Reply:
Opened https://github.com/scverse/anndata/issues/1392 to track